lisa/tools/kdump.py

```diff
@@ -0,0 +1,155 @@
+            if self.node.os.package_exists("linux-crashdump"):
+                return True
+        elif isinstance(self.node.os, Suse):
+            if (
+                self.node.os.package_exists("kexec-tools")
+                and self.node.os.package_exists("kdump")
+                and self.node.os.package_exists("makedumpfile")
+            ):
+                return True
+        return False
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            self.node.os.install_packages("kexec-tools")
+        elif isinstance(self.node.os, Debian):
+            self.node.os.install_packages("linux-crashdump")
+        elif isinstance(self.node.os, Suse):
+            package_names = ["kexec-tools", "kdump", "makedumpfile"]
+            self.node.os.install_packages(package_names)
+        return self._check_exists()
+
+    def config_kdump(
+        self,
+        dump_path: str,
+    ) -> None:
+        sed = self.node.tools[Sed]
+        echo = self.node.tools[Echo]
+        if isinstance(self.node.os, Redhat):
+            # Edit the /etc/kdump.conf file and specify the dump path
+            sed.replace("^path", "path", "#path", "/etc/kdump.conf", sudo=True)
+            echo.run(f"path {dump_path} > /etc/kdump.conf", shell=True, sudo=True)
+            # When kdump fails to create a core dump at the target location,
+            # kdump reboots the system without saving the vmcore
+            sed.replace("^default", "default", "#default", "/etc/kdump.conf", sudo=True)
+            echo.run("default reboot > /etc/kdump.conf", shell=True, sudo=True)
+        elif isinstance(self.node.os, Debian):
+            # Don't load a kexec kernel
+            sed.replace(
+                searched="",
+                original="LOAD_KEXEC=true",
+                replaced="LOAD_KEXEC=false",
+                file="/etc/default/kexec",
+                sudo=True,
+            )
+            # Enable kdump
+            sed.replace(
+                searched="",
+                original="USE_KDUMP=0",
+                replaced="USE_KDUMP=1",
+                file="/etc/default/kdump-tools",
+                sudo=True,
+            )
+
+        # Cleaning up any previous crash dump files
+        self.node.execute(
+            "mkdir -p /var/crash && rm -rf /var/crash/*", shell=True, sudo=True
+        )
+
+    def config_crashkernel_memory(
+        self,
+        crashkernel_memory: str,
+    ) -> None:
+        # For Ubuntu, /etc/default/grub.d/kdump-tools.cfg which is in kdump-tools package,
+        # will override the crashkernel config defined as GRUB_CMDLINE_LINUX_DEFAULT in /etc/default/grub
+        # We can keep the default crashkernel config for Ubuntu distro, no need to set crashkernel
+        posix_os: Posix = cast(Posix, self.node.os)
+        if isinstance(self.node.os, Redhat):
+            posix_os.config_kenel_cmdline(
+                "crashkernel=auto", f"crashkernel={crashkernel_memory}"
```

Comment:
Make sure "crashkernel=auto" works as a default value. If not, it's better to use regexp to replace it every time.

Reply:
my understanding is that we can remove all crashkernel value, and inject a new config, if we hard code the replaced value, then we have an assumption that 'crashkernel=auto' exists.